core/src/main/java/org/apache/iceberg/jdbc/JdbcUtil.java

```diff
@@ -303,6 +331,263 @@ public static Properties filterAndRemovePrefix(Map<String, String> properties, S
+            sql.setString(1, catalogName);
+            sql.setString(2, namespaceToString(tableOrViewIdentifier.namespace()));
+            sql.setString(3, tableOrViewIdentifier.name());
+            ResultSet rs = sql.executeQuery();
+
+            if (rs.next()) {
+              tableOrView.put(JdbcUtil.CATALOG_NAME, rs.getString(JdbcUtil.CATALOG_NAME));
+              tableOrView.put(
+                  (table ? JdbcUtil.TABLE_NAMESPACE : JdbcUtil.VIEW_NAMESPACE),
+                  rs.getString((table ? JdbcUtil.TABLE_NAMESPACE : JdbcUtil.VIEW_NAMESPACE)));
+              tableOrView.put(
+                  (table ? JdbcUtil.TABLE_NAME : JdbcUtil.VIEW_NAME),
+                  rs.getString((table ? JdbcUtil.TABLE_NAME : JdbcUtil.VIEW_NAME)));
+              tableOrView.put(
+                  BaseMetastoreTableOperations.METADATA_LOCATION_PROP,
+                  rs.getString(BaseMetastoreTableOperations.METADATA_LOCATION_PROP));
+              tableOrView.put(
+                  BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP,
+                  rs.getString(BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP));
+            }
+
+            rs.close();
+          }
+
+          return tableOrView;
+        });
+  }
+
+  /**
+   * Insert a table or view in the catalog.
+   *
+   * @param table true to insert a table, false to insert a view.
+   * @param connections the {@link JdbcClientPool} to connect to the database.
+   * @param catalogName the catalog name.
+   * @param namespace the namespace.
+   * @param tableOrViewIdentifier the table or view identifier.
+   * @param newMetadataLocation the new metadata location.
+   * @return the number of updated records.
+   * @throws SQLException if a SQL error happens on the database.
+   * @throws InterruptedException interrupt the SQL query execution.
+   */
+  public static int doCommitCreateTableOrView(
+      boolean table,
+      JdbcClientPool connections,
+      String catalogName,
+      Namespace namespace,
+      TableIdentifier tableOrViewIdentifier,
+      String newMetadataLocation)
+      throws SQLException, InterruptedException {
+    return connections.run(
+        conn -> {
+          try (PreparedStatement sql =
+              conn.prepareStatement(doCommitCreateTableOrViewSqlStatement(table))) {
+            sql.setString(1, catalogName);
+            sql.setString(2, namespaceToString(namespace));
+            sql.setString(3, tableOrViewIdentifier.name());
+            sql.setString(4, newMetadataLocation);
+            return sql.executeUpdate();
+          }
+        });
+  }
+
+  static boolean viewExists(
+      String catalogName, JdbcClientPool connections, TableIdentifier viewIdentifier) {
+    if (exists(
+        connections,
+        JdbcUtil.getTableOrViewSqlStatement(false),
+        catalogName,
+        namespaceToString(viewIdentifier.namespace()),
+        viewIdentifier.name())) {
+      return true;
+    }
+    return false;
```

Comment:
nit: Iceberg follows a codestyle of having an empty line after the block (If, for loop etc)